typed-racket-lib/typed-racket/types/abbrev.rkt

```diff
@@ -264,3 +264,15 @@
 (define-syntax-rule (-object . ?clauses)
   (make-Instance (-class . ?clauses)))
 
+
+(define-syntax (make-refine stx)
```

Comment:
There is not regular constructor of this form. The exposed constructor for intersections just intersects a bunch of types really simply. this provides a nice constructor to add a refinement to an arbitrary type, and then the helpers in `type-rep.rkt` can worry about what needs to be done to do this.